op-batcher/enclave-entrypoint.bash

```diff
@@ -108,36 +108,50 @@ filtered_args=()
 url_args=()
 
 SOCAT_PORT=10001
+echo "Arguments: $@"
 # Process all arguments
 while [ $# -gt 0 ]; do
+    echo "Processing argument: $1"
     # Check if the argument matches the URL pattern
-    if [[ $1 =~ $URL_ARG ]]; then
-        # Extract the flag part and possible value part
-        flag=${BASH_REMATCH[1]}
-
-        if [ $# -gt 1 ]; then
-            shift
-            value="$1"
-        else
-          echo "$flag doesn't have a value"
-          exit 1
-        fi
+    if [[ $1 =~ $URL_ARG_RE ]]; then
+      echo "Found URL argument: $1"
+      # Extract the flag part and possible value part
+      flag=${BASH_REMATCH[1]}
+
+      # extract value from "--flag=value" or "--flag value"
+      if [[ "$1" == *=* ]]; then
+        value="${1#*=}"
+      else
+        shift || { echo "$flag missing value"; exit 1; }
+        value="$1"
+      fi
 
-        echo "Rewriting $flag=$value"
+      # SPECIAL CASE: espresso-url may be a comma list. Either split here,
+      # or (simpler) pass the flag twice at build time. See note below.
+      if [[ "$flag" == "--espresso-url" && "$value" == *","* ]]; then
+        IFS=',' read -r -a parts <<< "$value"
```

Comment:
Can't we just do this for all values without treating this as a special case? Looks like this code will handle singular values without commas the same way it handles them currently.